roles/postgresql_upgrade/tasks/main.yml

```diff
@@ -0,0 +1,62 @@
+---
+- name: Get version of postgres data directory
```

Comment:
I thought more about the following:

```yaml
- name: Ensure running on EL8
  assert:
    that:
      - ansible_os_family == "RedHat"
      - ansible_distribution_major_version == '8'
    fail_msg: "This role is only supported on EL8"

- name: Get version from the PostgreSQL data directory
  slurp:
    path: /var/lib/pgsql/data/PG_VERSION
  register: pg_version

- name: Ensure PostgreSQL version is 10
  assert:
    that:
      - "'10' in pg_version['content']|b64decode"
    fail_msg: "This role only supports upgrades from PostgreSQL 10"

[ rest of the tasks without the block ]
```

But wouldn't block on my personal preferences ;)

Reply:
The initial assert makes sense to me. The last assert will break idempotency I think?

Reply:
True.

Reply:
True.